pkg/cli/declarative_print_rules.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/cli/clierrorplus"
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scplan"
+	"github.com/cockroachdb/errors"
+	"github.com/spf13/cobra"
+)
+
+var declarativePrintRules = &cobra.Command{
+	Use:   "declarative-print-rules <version> <dep|op>",
+	Short: "validates a corpus file for the declarative schema changer",
+	Long: `
+Validates very single declarative schema changer state can be planned against in
+a given corpus file.
+`,
+	Args: cobra.ExactArgs(2),
+	RunE: clierrorplus.MaybeDecorateError(
+		func(cmd *cobra.Command, args []string) (resErr error) {
+			ctx := context.Background()
+			version, err := roachpb.ParseVersion(args[0])
+			if err != nil {
+				return err
+			}
+			rules := scplan.GetRulesRegistryForRelease(ctx,
```

Comment:
maybe return a list of available instead of just falling back to the oldest rules if the given version is bad?

Reply:
I added support for dumping version numbers if an invalid one is used.